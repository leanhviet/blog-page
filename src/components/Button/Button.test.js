// Libs
import React from 'react'
import { shallow } from 'enzyme'

// Components
import { ButtonBase } from './index'

const mockFn = jest.fn()
const props = {
  onClick: mockFn,
  label: 'Create a blog',
  className: 'btn',
}

describe('Button component', () => {
  const wrapper = shallow(<ButtonBase />)
  const wrapperWithProps = shallow(<ButtonBase {...props} />)

  it('should be defined Button component', () => {
    expect(wrapper).toBeDefined()
  })

  // Testing Snapshot
  it('should be render snapshot Button component', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapperWithProps).toMatchSnapshot()
  })

  // Testing default props
  it('should be render defaultProps Button component', () => {
    expect(ButtonBase.defaultProps.className).toEqual('')
    expect(ButtonBase.defaultProps.label).toEqual('')
    expect(ButtonBase.defaultProps.onClick()).toBeUndefined()
  })

  // Testing mock props
  it('should be render mockProps Button component', () => {
    expect(wrapperWithProps.find('button').props().className).toEqual(
      props.className,
    )
    expect(wrapperWithProps.find('button').props().children).toEqual(
      props.label,
    )
  })

  // Testing for action click
  it('behavior Button component', () => {
    wrapperWithProps.simulate('click', {})
    expect(mockFn).toHaveBeenCalledTimes(1)
  })
})
