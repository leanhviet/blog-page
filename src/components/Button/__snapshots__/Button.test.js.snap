// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should be render snapshot Button component 1`] = `
<button
  className=""
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;

exports[`Button component should be render snapshot Button component 2`] = `
<button
  className="btn"
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  Create a blog
</button>
`;
